Description: do not unregister the agent under GNOME Flashback
 GNOME Flashback owns the org.gnome.Shell name, but does not (currently)
 export ShellVersion, which is causing problems in nm-applet.
Author: Dmitry Shachnev <mitya57@ubuntu.com>
Forwarded: not-needed
Last-Update: 2015-04-03

--- a/src/applet.c
+++ b/src/applet.c
@@ -4113,6 +4113,28 @@
 	gtk_status_icon_set_visible (applet->status_icon, applet->visible);
 }
 
+static gboolean is_gnome_flashback ()
+{
+	const gchar *xdg_current_desktop;
+	gchar **desktop_names;
+	int i;
+	gboolean result;
+
+	result = FALSE;
+	xdg_current_desktop = g_getenv ("XDG_CURRENT_DESKTOP");
+	if (xdg_current_desktop != NULL) {
+		desktop_names = g_strsplit (xdg_current_desktop, ":", 0);
+		for (i = 0; desktop_names[i]; ++i) {
+			if (!g_strcmp0 (desktop_names[i], "GNOME-Flashback")) {
+				result = TRUE;
+				break;
+			}
+		}
+		g_strfreev (desktop_names);
+	}
+	return result;
+}
+
 static gboolean
 initable_init (GInitable *initable, GCancellable *cancellable, GError **error)
 {
@@ -4221,7 +4243,7 @@
 	applet_embedded_cb (G_OBJECT (applet->status_icon), NULL, NULL);
 
 	/* Watch GNOME Shell so we can unregister our applet agent if it appears */
-	if (!shell_debug) {
+	if (!shell_debug && !is_gnome_flashback ()) {
 		applet->shell_watcher = nm_shell_watcher_new ();
 		g_signal_connect (applet->shell_watcher,
 			              "notify::shell-version",
